src/main/java/io/hops/hopsworks/expat/migrations/models/OpenSearchToRonDBMigration.java

```diff
@@ -0,0 +1,280 @@
+      .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+      .build();
+    this.expatModelsController = new ExpatModelsController(this.connection);
+    this.expatInodeController = new ExpatInodeController(this.connection);
+    this.expatProjectFacade = new ExpatProjectFacade(ExpatProject.class, this.connection);
+    this.expatUserFacade = new ExpatUserFacade();
+    this.expatHdfsUserFacade = new ExpatHdfsUserFacade(ExpatHdfsUser.class, this.connection);
+  }
+
+  private void close() throws SQLException, IOException {
+    if(connection != null) {
+      connection.close();
+    }
+    if(httpClient != null) {
+      httpClient.close();
+    }
+  }
+
+  @Override
+  public void migrate() throws MigrationException {
+    //TODO make sure elastic does not limit number of returned responses
+    try {
+      setup();
+      LOGGER.info("Getting all file provenance indices");
+      JSONArray fileProvIndices = ElasticClient.getIndicesByRegex(httpClient, elastic, elasticUser, elasticPass,
+        "*__file_prov");
+      LOGGER.info(fileProvIndices.toString());
+      if (fileProvIndices.length() > 0) {
+        LOGGER.info("Found {} file provenance indices to migrate", fileProvIndices.length());
+        for(int i = 0; i < fileProvIndices.length(); i++) {
+          JSONObject fileProvIndexObj = fileProvIndices.getJSONObject(i);
+          String fileProvIndexName = fileProvIndexObj.getString("index");
+          long projectInodeId = Long.parseLong(fileProvIndexName.substring(0, fileProvIndexName.indexOf("__")));
+
+          ExpatHdfsInode projectInode = expatInodeController.getInodeById(projectInodeId);
+          String projectName = projectInode.getName();
+          ExpatHdfsInode modelDatasetInode = expatInodeController.getInodeAtPath(
+            String.format("/Projects/%s/Models", projectName));
+
+          String query = "{\"from\":0,\"size\":10000,\"query\":{\"bool\":{\"must\":[{\"term\":{\"entry_type\":" +
+            "{\"value\":\"state\",\"boost\":1.0}}},{\"bool\":{\"should\":[{\"term\":{\"project_i_id\":" +
+            "{\"value\":\"" + projectInode.getId() + "\",\"boost\":1.0}}}]" +
+            ",\"adjust_pure_negative\":true,\"boost\":1.0}},{\"bool\":" +
+            "{\"should\":[{\"term\":{\"ml_type\":{\"value\":\"MODEL\",\"boost\":1.0}}}]," +
+            "\"adjust_pure_negative\":true,\"boost\":1.0}},{\"bool\":{\"should\":[{\"term\":{\"dataset_i_id\":" +
+            "{\"value\":\"" + modelDatasetInode.getId() +"\",\"boost\":1.0}}}]" +
+            ",\"adjust_pure_negative\":true,\"boost\":1.0}},{\"exists\":" +
+            "{\"field\":\"xattr_prov.model_summary.value\",\"boost\":1.0}}]," +
+            "\"adjust_pure_negative\":true,\"boost\":1.0}}}";
+
+          JSONObject resp = ElasticClient.search(httpClient, elastic, elasticUser, elasticPass, fileProvIndexName,
+            query);
+
+          JSONArray modelHits = resp.getJSONObject("hits").getJSONArray("hits");
+          if(modelHits.length() > 0) {
```

Comment:
i would add a check above between `hits.length` and `hits.total` and just abort if the two don't match.
this should be an extreme case where a project has more than 10000 models. Realistically it should not happen, but it would be good to have a fail fast check, just in case.

Reply:
Added a check here https://github.com/logicalclocks/expat/pull/116/commits/6570b9ca5e13fa329c189f63e778dec42c09adcb